lib/models/playlist.js

```diff
@@ -32,11 +32,51 @@ const destroy = (playlistId) => database('playlists')
   .where('id', playlistId)
   .del()
 
+const find = (id) => database('playlists').where("id", id).first()
+
+const getPlaylistFavorites = (playlistId) => database('playlist_favorites')
+  .select('favorites.id', 'favorites.title', 'favorites.artistName', 'favorites.genre', 'favorites.rating')
+  .join('favorites', { 'favorites.id': 'playlist_favorites.favoriteId'})
+  .where({ playlistId: playlistId })
+  .groupBy('favorites.id')
+
+const getAvgRating = (playlistId) => database('playlist_favorites')
+  .join('favorites', { 'favorites.id': 'playlist_favorites.favoriteId'})
+  .where({ playlistId: playlistId })
+  .avg('favorites.rating as songAvgRating')
```

Comment:
This section is returning null when there are no favorites for a playlist. Try using `leftJoin` instead of just `join`

Reply:
Excellent catch. I fixed it by adding conditionals in the return object being serialized.